cranelift/codegen/src/machinst/compile.rs

```diff
@@ -95,3 +99,137 @@ pub fn compile<B: LowerBackend + TargetIsa>(
 
     Ok((vcode, regalloc_result))
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Thanks for adding these tests, but (i) they all appear to be testing `is_leaf`, which we want to remove, per above; (ii) they don't actually test the ABI behavior, which is visible only in the emitted code.

Could you add some compile filetests instead, showing disassemblies with and without frame-pointer setup? In particular I'd expect the case where I commented and asked for #11573 to be filed to revert back to no-frame-pointer.